core/deployment/src/main/java/io/quarkus/deployment/util/WebJarUtil.java

```diff
@@ -0,0 +1,174 @@
+package io.quarkus.deployment.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import io.quarkus.bootstrap.model.AppArtifact;
+import io.quarkus.bootstrap.model.AppDependency;
+import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
+
+/**
+ * Utility for Web resource related operations
+ */
+public class WebJarUtil {
+
+    private final static String tmpDir = System.getProperty("java.io.tmpdir");
+    private static final String OWN_MEDIA_FOLDER = "META-INF/branding/";
+    private static final List<String> IGNORE_LIST = Arrays.asList("logo.png", "favicon.ico");
```

Comment:
Do we see this changing with updates of UI? Maybe having it as property will be better

Reply:
Nope. This is added manually by us in SmallRye. All 3 of the current UI's. In SmallRye it's the SmallRye logo. There is a 3rd one coming in later releases, style.css. That will allows users to do better custom branding